packages/ember-data/lib/system/relationship-meta.js

```diff
@@ -8,7 +8,7 @@ export function typeForRelationshipMeta(store, meta) {
     if (meta.kind === 'hasMany') {
       typeKey = singularize(typeKey);
     }
-    type = store.modelFor(typeKey);
```

Comment:
`store` is now no longer needed as an argument to this function. which means  `relatedTypesDescriptor`  and `relationshipsByNameDescriptor` should no longer need the store either.